src/highdicom/seg/sop.py

```diff
@@ -940,3 +962,1491 @@ def _encode_pixels(self, planes: np.ndarray) -> bytes:
+        in a list of tuples.
+
+        Note that the order of the returned list is not significant and will
+        not in general match the order of segments.
+
+        Parameters
+        ----------
+        segmented_property_category: Optional[Union[Code, CodedConcept]], optional
+            Segmented property category filter to apply.
+        segmented_property_type: Optional[Union[Code, CodedConcept]], optional
+            Segmented property type filter to apply.
+        algorithm_type: Optional[Union[SegmentAlgorithmTypeValues, str]], optional
+            Segmented property type filter to apply.
+
+        Returns
+        -------
+        List[Tuple[str, UID]]
+            List of all unique (tracking_id, tracking_uid) tuples that are
+            referenced in segment descriptions in this SEG image that match
+            all provided filters.
+
+        """  # noqa: E501
+        filter_funcs = []
+        if segmented_property_category is not None:
+            filter_funcs.append(
+                lambda desc:
+                desc.segmented_property_category == segmented_property_category
+            )
+        if segmented_property_type is not None:
+            filter_funcs.append(
+                lambda desc:
+                desc.segmented_property_type == segmented_property_type
+            )
+        if algorithm_type is not None:
+            algo_type = SegmentAlgorithmTypeValues(algorithm_type)
+            filter_funcs.append(
+                lambda desc:
+                SegmentAlgorithmTypeValues(desc.algorithm_type) == algo_type
+            )
+
+        return list({
+            (desc.tracking_id, UID(desc.tracking_uid))
+            for desc in self.SegmentSequence
+            if desc.tracking_id is not None and
+            all(f(desc) for f in filter_funcs)
+        })
+
+    @property
+    def segmented_property_categories(self) -> List[CodedConcept]:
+        """Get all unique segmented property categories in this SEG image.
+
+        Returns
+        -------
+        List[CodedConcept]
+            All unique segmented property categories referenced in segment
+            descriptions in this SEG image.
+
+        """
+        categories = []
+        for desc in self.SegmentSequence:
+            if desc.segmented_property_category not in categories:
+                categories.append(desc.segmented_property_category)
+
+        return categories
+
+    @property
+    def segmented_property_types(self) -> List[CodedConcept]:
+        """Get all unique segmented property types in this SEG image.
+
+        Returns
+        -------
+        List[CodedConcept]
+            All unique segmented property types referenced in segment
+            descriptions in this SEG image.
+
+        """
+        types = []
+        for desc in self.SegmentSequence:
+            if desc.segmented_property_type not in types:
+                types.append(desc.segmented_property_type)
+
+        return types
```

Comment:
```suggestion
        return [
            desc.segmented_property_type
            for desc in self.SegmentSequence
            if desc.segmented_property_type not in types
        ]
```

Reply:
This can't be written as a list comprehension because the check is applied to the list that is being constructed. It appends the segmented property to the list only if it is not already in the list.

Reply:
Sorry...didn't look carefully enough at it.